light/write.cc

```diff
@@ -844,11 +911,19 @@ void SaveLightmapSurface(const mbsp_t *bsp, mface_t *face, facesup_t *facesup,
         lux = lux_filebase.data() + (id.lightofs * 3);
     }
 
+    if (!hdr_filebase.empty()) {
+        hdr = hdr_filebase.data() + (id.lightofs * 4);
+    }
+
     int lightofs;
     
     // Q2/HL native colored lightmaps
     if (bsp->loadversion->game->has_rgb_lightmap) {
-        lightofs = lit - lit_filebase.data();
+        if (light_options.write_litfile & lightfile::hdr) {
```

Comment:
I think this change is wrong; the `lightofs` is setting the offset in the lighting data of the start of the face's lightmap, but it should be in units of the native lightmap format of the bsp.

So for Q1BSP `lightofs` is always in units of 1 byte per luxel (even if we're also writing extended lits/bspx lumps with more bytes-per-luxel), and for Q2BSP and HLBSP it's always assuming 3 bytes per luxel (again even when writing supplemental lumps.)

(`has_rgb_lightmap==true` means the BSP's native lightmap format is 3 bytes per luxel, `false` means 1 byte per luxel).

Reply:
Ah, right, I had it lilke that from the start like original PR, but changed when I had issues with relighting official dm2.bsp, but that was due to temp-files from a case where I had compiled dm2.bsp from .map and there was a mismatch. Forgot to change that back.